services/src/main/java/org/keycloak/services/managers/DefaultBruteForceProtector.java

```diff
@@ -261,125 +152,54 @@ protected void sendEvent(KeycloakSession session, RealmModel realm, UserLoginFai
-                                } else if (event instanceof SuccessfulLogin) {
-                                    ((SuccessfulLogin) event).latch.countDown();
-                                }
-                            }
-                            events.clear();
-                        }
-                    } catch (Exception e) {
-                        ServicesLogger.LOGGER.failedProcessingType(e);
-                    }
-                } catch (InterruptedException e) {
-                    break;
-                }
-            }
-        } finally {
-            shutdownLatch.countDown();
-        }
-    }
-
-    protected void success(KeycloakSession session, LoginEvent event) {
-        String userId = event.userId;
+    public void shutdown() {}
 
-        UserLoginFailureModel user = getUserModel(session, event);
-        if(user == null) return;
+    protected void success(KeycloakSession session, RealmModel realm, String userId) {
+        UserLoginFailureModel userLoginFailure = getUserFailureModel(session, realm, userId);
+        if(userLoginFailure == null) return;
         if (logger.isDebugEnabled()) {
-            UserModel model = session.users().getUserById(getRealmModel(session, event), userId);
+            UserModel model = session.users().getUserById(realm, userId);
             logger.debugv("user {0} successfully logged in, clearing all failures", model.getUsername());
         }
-        user.clearFailures();
-    }
-
-    protected void logFailure(LoginEvent event) {
-        failures++;
-        long delta = 0;
-        if (lastFailure > 0) {
-            delta = Time.currentTimeMillis() - lastFailure;
-            if (delta > (long)maxDeltaTimeSeconds * 1000L) {
-                totalTime = 0;
-
-            } else {
-                totalTime += delta;
-            }
-        }
+        userLoginFailure.clearFailures();
     }
 
     @Override
     public void failedLogin(RealmModel realm, UserModel user, ClientConnection clientConnection) {
-        try {
-            FailedLogin event = new FailedLogin(realm.getId(), user.getId(), clientConnection);
-            queue.offer(event);
-            // wait a minimum of seconds for type to process so that a hacker
-            // cannot flood with failed logins and overwhelm the queue and not have notBefore updated to block next requests
-            // todo failure HTTP responses should be queued via async HTTP
-            event.latch.await(5, TimeUnit.SECONDS);
-        } catch (InterruptedException e) {
-        }
+        processLogin(realm, user, clientConnection, false);
+        // wait a minimum of seconds for type to process so that a hacker
+        // cannot flood with failed logins and overwhelm the queue and not have notBefore updated to block next requests
+        // todo failure HTTP responses should be queued via async HTTP
+        //event.latch.await(5, TimeUnit.SECONDS);
```

Comment:
Maybe we should delete these comments. My feeling this was important in the past because maybe the success/failure was executed in the same thread and not off-thread like it is now. But this is a minor thing. Approving anyway.

Reply:
+1, but yeah, likely does not worth to wait even more for this PR just because of it.

Maybe we can delete during some next PR, which need to do any changes in brute-force protector.